README.md

```diff
@@ -1,108 +1,209 @@
+
+Any choice brings significant disadvantages:
+
+- If such features are included as features that are mandatory to support,
+  some providers may have difficulties adopting the API.
+- If such features are included as optional features, the ability to migrate
+  from one provider to another suffers significantly. Without this ability,
+  users also may opt to use provider-specific API's, instead.
+- If such features are excluded, the API becomes overall less useful for the
+  users who may opt to use more powerful provider-specific API's, instead.
+
+In sum: Going this route would be technically the best thing to do, yet does
+not seem feasible given tough trade-offs and limited resources.  
+If the opportunity arises to partner with some other organization with a lot
+of staff and resources, this option may be reevaluated, though.
+
+</details>
+
+## The chosen approach (for now)
+
+Goal: **Provide a "semantically" consistent API modelling most cloud resources
+that are in scope for SCS**.
+
+### Kubernetes API
+
+Instead of creating SCS-specific API idioms and building the implementation
+from scratch, the Kubernetes API will be "reused". Essentially, the Kubernetes
+API is just an opinionated REST API which has opinions on how a resource
+is defined, how it looks like, how it is reconciled/handled, how AuthN/AuthZ
+can be implemented. The Kubernetes ecosystem provides much tooling for working
+with such (custom) resource definitions: For creating the definitions
+themselves, building controllers, making them discoverable and deployable.
+
+As such, Kubernetes is a great choice for building any sort of resource
+management API - with some caveats regarding its deployment and the legacy
+of starting off as container orchestration tooling.
+
+### Crossplane tooling
+
+Crossplane even extends the Kubernetes API with
+"[Compositions](https://docs.crossplane.io/v1.14/concepts/compositions/)" and
+"[Composite Resource Definitions](https://docs.crossplane.io/v1.14/concepts/composite-resource-definitions/)"
+(XRD) to make Kubernetes the base for platform engineering within organizations.
+
+Secondly, it provides an API machinery to bring any cloud resource into Kubernetes
+using backend-specific "providers" (roughly comparable with Terraform providers).
+As such, Crossplane with its provider ecosystem actually already did most of
+the heavy lifting for providing e.g. OpenStack resources inside of Kubernetes.
+
+On top, the platform engineering concepts in Crossplane make building multi-tenancy
+systems pretty straight-forward, even for
+[single clusters](https://docs.crossplane.io/knowledge-base/guides/multi-tenant/#single-cluster-multi-tenancy).
+
+Alright. Crossplane takes care of exposing OpenStack resources and does some
+fancy stuff regarding multi-tenancy. What about providing actual Kubernetes
+**workload** clusters?
+
+### Cluster stacks
+
+Cluster stacks are basically an abstraction over Cluster API. As such, Cluster
```

Comment:
I agree with @mxmxchere that this part has to be written better. I would in general and very high-level say that Cluster Stacks make it easier to work with Cluster API with various features, like tested upgrades, creating n clusters from the same template, etc.

Reply:
As in @mxmxchere's review [above](https://github.com/SovereignCloudStack/central-api/pull/4#discussion_r1484147650), thanks for clarifying. Will change that wording.

Reply:
Wording is changed.